### PR TITLE
LNK-2106

### PR DESCRIPTION
**Pull request by nvmLantana**
Move Exception Handling into Shared

### CODE REVIEW

File: Shared/Application/Error/Handlers/TransientExceptionHandler.cs
Comment (edward-miller-lcg) [inline Shared/Application/Error/Handlers/TransientExceptionHandler.cs:36]: Would it make sense to pass the topic name as a method argument here and pass it down as opposed to setting it in the calling class?

**Review by edward-miller-lcg — COMMENTED** [review]
@dvargaslantana
I don't see NonTransientException handling implemented for deadletter workflow so that would need to be added either before the PR is approved or as an after task.

**Comment by MontaltoNick** [comment]
> @dvargaslantana I don't see NonTransientException handling implemented for deadletter workflow so that would need to be added either before the PR is approved or as an after task.

ExceptionHandler is the workflow for the 'non transient' exception case. It just logs the audit message, and back in the service we mark the event as processed so it is removed from the queue.

**Comment by edward-miller-lcg** [comment]
I think we should be putting that in a specific non transient case that produces a dead letter. I don't understand why we would not do that for NonTransient but _do_ that for Transient cases.

**Review by edward-miller-lcg — APPROVED** [review]

**Pull request merged**
